### PR TITLE
VisIt: Update for building with 3.2.1.

### PR DESCRIPTION
**Pull request by brugger1**
I updated the VisIt package to build with 3.2.1.

It includes several several patches to vtk 8.1 for building on a system with no system install X11 libraries or include files.

It specifies specific versions of dependent packages that are known to work with 3.2.1.

It was tested on spock.olcf.ornl.gov. The GUI came up and rendered images and an image was successfully saved using off screen rendering from data from curv2d.silo.

I had to make some changes to my environment on spock to get the spack package to work properly.

For pcre to build, I had to do:

LD_LIBRARY_PATH=/opt/cray/pe/gcc/10.3.0/snos/lib64:/opt/cray/libfabric/1.11.0.4.75/lib64

I also had to change:

extra_rpaths: []

to

extra_rpaths: [/opt/cray/libfabric/1.11.0.4.75/lib64]

for the gcc@10.3.0 compiler in .spack/cray/compilers.yaml

Finally for cairo to install I had to do:

export LC_ALL=en_US.utf8

I'm not sure how to fix these for a user in general, but I am documenting them here.

### CODE REVIEW

**Review by bernhardkaindl — DISMISSED** [review]
I confirm, it builds with %gcc@10.3.0 on Ubuntu-18.04.

GCC from: https://launchpad.net/~ubuntu-toolchain-r/+archive/ubuntu/ppa - installed as follows
```
sudo apt-get install -y software-properties-common
sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
sudo apt-get update
sudo apt-get install -y g++-10 gfortran-10 
spack compiler find
```
No other tweaks were needed. A few notes or hits for improvements:
* spack maintainers might not like removing old versions (even despite them not building ;-)
* The landing page https://wci.llnl.gov/simulation/computer-codes/visit is actually quite a nice intro and has the link to the github page, so it could be kept as well.
* I think the vtk patches would deserve comments about what they fix
* Would the patches for vtk helpful for the vtk package itself? In the past I had no issues building vtk@9.0.3 in Ubuntu clouldimg-style containers without X11.
* It would be nicer if the patches would be made in the unified format `diff -u` which is the modern diff/patch standard.

File: var/spack/repos/builtin/packages/visit/package.py
Comment (bernhardkaindl) [inline var/spack/repos/builtin/packages/visit/package.py]: I think, which version of LLVM Mesa uses to build itself should be a private topic of Mesa:
I think it should be free to be compiled with whatever versions Mesa supports.

File: var/spack/repos/builtin/packages/visit/package.py
Comment (bernhardkaindl) [inline var/spack/repos/builtin/packages/visit/package.py]: I guess it would be much nicer for building no versions would be pinned, but if you'd give a range of versions.

If your time permits, could you also test with these versions set as minimum versions - for allowing spack to use later or the lastest versions of these packages? (and of qwt above too).

Please take this just as an review from an outsider. For the maintainer-review I am told to wait for 7 days.

**Comment by brugger1** [comment]
I'll make the improvements suggested, so hold off on merging until I indicate I'm finished.

File: var/spack/repos/builtin/packages/visit/package.py
Comment (brugger1) [reply var/spack/repos/builtin/packages/visit/package.py]: I pinned LLVM to versions "6:8". Newer versions did not work with older compilers that we still use.

File: var/spack/repos/builtin/packages/visit/package.py
Comment (brugger1) [reply var/spack/repos/builtin/packages/visit/package.py]: I removed the pinning where I could and loosened it up where I could. I added comments explaining why things were pinned.

**Comment by brugger1** [comment]
I believe I addressed all the review comments.
I put back in the old versions.
I put back in the old homepage.
I added comments explaining what each VTK patch was doing. It may be that VTK9 doesn't need any of these patches, but we haven't completed our upgrade to VTK9.
I changed the patches to use "diff -u".
I didn't try to submit the VTK patches to the VTK spack package. I may do that in the future, but since they are for an older version of VTK, perhaps they aren't that important.

**Comment by scottwittenburg** [comment]
@spackbot run pipeline

**Review by bernhardkaindl — APPROVED** [review]
Built and launched fine.

**Review by cyrush — APPROVED** [review]

**Pull request merged**
